tt_metal/impl/device/device_mesh_view.cpp

```diff
@@ -76,7 +76,7 @@ DeviceMeshView::DeviceView DeviceMeshView::get_devices(const Coordinate& start,
 }
 
 DeviceMeshView::DeviceView DeviceMeshView::get_devices(const DeviceGrid& shape) {
-    return get_devices({0, 0}, {shape.first - 1, shape.second - 1});
+    return get_devices({0, 0}, {(size_t)(shape.first - 1), (size_t)(shape.second - 1)});
```

Comment:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast